src/intersection.class.ts

```diff
@@ -1,166 +1,226 @@
 //@ts-nocheck
-import { Point } from './point.class';
+import { Point } from "./point.class";
+import { fabric } from '../HEADER';
 
-(function(global) {
-  /* Adaptation of work of Kevin Lindsey (kevin@kevlindev.com) */
-  var fabric = global.fabric || (global.fabric = { });
-  /**
-   * Intersection class
-   * @class fabric.Intersection
-   * @memberOf fabric
-   * @constructor
-   */
-  function Intersection(status) {
+/* Adaptation of work of Kevin Lindsey (kevin@kevlindev.com) */
+
+type IntersectionType = 'Intersection' | 'Coincident' | 'Parallel';
+
+/**
+ * **Assuming `T`, `A`, `B` are points on the same line**,
+ * check if `T` is contained in `[A, B]` by comparing the direction of the vectors from `T` to `A` and `B`
+ * @param T 
+ * @param A 
+ * @param B 
+ * @returns true if `T` is contained
+ */
+const isContainedInInterval = (T: Point, A: Point, B: Point) => {
+  const TA = new Point(T).subtract(A);
+  const TB = new Point(T).subtract(B);
+  return Math.sign(TA.x) !== Math.sign(TB.x) || Math.sign(TA.y) !== Math.sign(TB.y);
+}
+
+export class Intersection {
+
+  points: Point[]
+
+  status?: IntersectionType
+
+  constructor(status?: IntersectionType) {
     this.status = status;
     this.points = [];
   }
 
-  fabric.Intersection = Intersection;
-
-  fabric.Intersection.prototype = /** @lends fabric.Intersection.prototype */ {
-
-    constructor: Intersection,
-
-    /**
-     * Appends a point to intersection
-     * @param {Point} point
-     * @return {fabric.Intersection} thisArg
-     * @chainable
-     */
-    appendPoint: function (point) {
-      this.points.push(point);
-      return this;
-    },
-
-    /**
-     * Appends points to intersection
-     * @param {Array} points
-     * @return {fabric.Intersection} thisArg
-     * @chainable
-     */
-    appendPoints: function (points) {
-      this.points = this.points.concat(points);
-      return this;
-    }
-  };
+  /**
+   * 
+   * @param {Point} point 
+   * @returns 
+   */
+  contains(point) {
+    return this.points.some(p => p.eq(point));
+  }
 
   /**
-   * Checks if one line intersects another
-   * TODO: rename in intersectSegmentSegment
+   * Appends points of intersection
+   * @param {...Point[]} points
+   * @return {Intersection} thisArg
+   * @chainable
+   */
+  private append(...points) {
```

Comment:
refactored `appendPoint` `appendPoints` to `append` es6!
I don't consider it breaking as this method is private